routers/maze.js

```diff
@@ -19,55 +22,21 @@ const findValue = (maze, operValue) => {
   return nums;
 };
 
-const validateArray = maze => {
-  maze.forEach(x => {
-    if (Array.isArray(x)) {
-      validateArray(x);
-    } else {
-      if (typeof x != 'number') {
-        throw new Error('Only numbers accepted');
-      }
-    }
-  });
-};
-
-router.post('/find', function (req, res) {
+router.get('/find', validateFindRequest, (req, res) => {
   try {
-    const { operator, value } = req.body;
-    let oper;
-    try {
-      if (operator === undefined || value === undefined) {
-        throw new Error(
-          'Request should contain "operator" and "value" properties'
-        );
-      }
-      if (typeof value != 'number') {
-        throw new Error('Value should be a number');
-      }
-      oper = getOperator(operator);
-    } catch (err) {
-      return res.status(400).json({ error: err.message });
-    }
-    const nums = findValue(maze, { operator: oper, value });
+    const nums = findValue(maze, req.operValue);
     return res.status(200).json({ result: nums.sort() });
   } catch (err) {
     return res.status(502).json({ error: err.message });
```

Comment:
5xx errors are usually return for serverside errors. All of these are client side errors.

Reply:
@zatziky The idea here is that 400 is sent in request validation and 502 is sent just in case something happens on the server (for example while finding value in recursion) which is a server error in my opinion.

Reply:
It kind of explains it. We were trying to understand with our tech head and we did not get it. 
But it makes sense. Yet, I would probably do it in error middleware to catch all of these unpredicted errors in one place. From there you could handle them uniformly.